document-portal/meson.build

```diff
@@ -0,0 +1,63 @@
+# Note: this can be flattened into the main meson.build file but the
```

Comment:
Folding `document-portal/meson.build` into `./meson.build` is not really desirable anyway: in Meson it's conventional to have one `meson.build` per directory, and that doesn't have the same disadvantages that it would in Autotools (Meson/ninja always behaves like non-recursive `make`, even when it's "recursive" at the `meson.build` level).

Reply:
The biggest issue I found with using `subdir()` in meson is that it only works as long as all components play along. For normal builds that's fine but gdbus-codegen produces multiple logically separate entities (in our case: code and docs) so using `subdir` means having to reach across directories and using variables defined in other `meson.build` files which is harder to read. Flattening things out can help with that, sometimes.